tests/trx_generator/trx_generator_tests.cpp

```diff
@@ -435,4 +488,46 @@ BOOST_AUTO_TEST_CASE(account_name_generator_tests)
    }
 }
 
+BOOST_AUTO_TEST_CASE(simple_http_client_async_test)
+{
+
+   auto const host = "127.0.0.1";
+   auto const port = 8888;
+   auto const port_str = "8888";
+
+   //Start Server
+   trx_gen_impl server = trx_gen_impl();
+   auto addr = boost::asio::ip::address::from_string(host);
+   boost::asio::ip::tcp::endpoint endpoint(addr, port);
+
+   server.start(endpoint);
+
+   //Start Client
+   auto const target = "/v1/chain/test";
+   int version = 11;
+   auto const content_type = "text/plain";
+   auto const test_body = "test request body";
+   auto const content_type2 = "application/json";
+   auto const test_body2 = "{\"return_failure_trace\":true,\"retry_trx\":false,\"transaction\":{\"signatures\":[\"SIG_K1_JyzLqbvpdybyujtiN1YdY2FWcBBi8dWWiFgZ515qyyqgKJJ6892i4rXTHdw5KGYut6EBuXPR3ExRwPSioSZ2bZ1RjNUXVj\"],\"compression\":\"none\",\"packed_context_free_data\":\"\",\"packed_trx\":\"848a34641800f994a24e00000000030000000000ea305500409e9a2264b89a0160ae423ad15b974a00000000a8ed32326660ae423ad15b974a1042088a4dd35057010000000100038d26b3d5ce8c7d76ef00d3d586a3d7bbc76c42f0b0719cc6f7b0cce1790622c301000000010000000100028dc3921705c71d30b0b26674536fff934f8e43890c980aa1d2c168f00f406539010000000000000000ea3055000000004873bd3e0160ae423ad15b974a00000000a8ed32322060ae423ad15b974a1042088a4dd35057009435770000000004535953000000000000000000ea305500003f2a1ba6a24a0160ae423ad15b974a00000000a8ed32323160ae423ad15b974a1042088a4dd3505740420f0000000000045359530000000040420f000000000004535953000000000000\"}}";
+
+   // The io_context is required for all I/O
+   net::io_context ioc;
+
+   // Launch the asynchronous operation
+   std::future<http::response<http::string_body>> future_response = std::make_shared<session>(ioc)->run(host, port_str, target, version, content_type, test_body);
```

Comment:
It's very clear to me what scenarios the http client code will be used; however, I prefer to use callback model than the std::future model. Furthermore, you can wrap this statement into a more readable function.